ios/MaasCore/Sources/MaasCore/Api/Utils/ApiKotlinFlowPublisher.swift

```diff
@@ -27,14 +27,21 @@ private extension ApiKotlinFlowPublisher {
             self.subscriber = subscriber
             
             self.closable = flow.watch { result in
-                if let success = result as? ApiResultSuccess<T> {
+                switch result {
+                case let success as ApiResultSuccess<T>:
                     let _ = subscriber.receive(success.value)
-                } else if let failure = result as? ApiResultFailure<T> {
-                    // TODO: failure.exception as ApiError ???
-                    subscriber.receive(completion: .failure(ApiError.apiResultFailure))
-                } else {
-                    let message = "ApiResult: unable to parse to ApiResultSuccess or ApiResultFailure, Expecting type: \(T.self)"
-                    subscriber.receive(completion: .failure(ApiError.error(message)))
+                case let failure as ApiResultFailureUnauthorized<T>:
+                    subscriber.receive(completion: .failure(ApiError.unauthorized(error: failure.error)))
+                case let failure as ApiResultFailureError<T>:
```

Comment:
Can we use this like that ? 🤔 
```suggestion
                    subscriber.receive(completion: .failure(.unauthorized(error: failure.error)))
                case let failure as ApiResultFailureError<T>:
```

Reply:
We sure can! Swift 😍